apps/web/src/components/balances/AssetsTable/index.tsx

```diff
@@ -316,9 +337,85 @@ const AssetsTable = ({
+                    <Box key={item.tokenInfo.address} className={css.mobileRow}>
+                      <Box className={css.mobileAssetRow}>
+                        <div className={css.token}>
+                          <TokenIcon logoUri={item.tokenInfo.logoUri} tokenSymbol={item.tokenInfo.symbol} />
+
+                          <Stack>
+                            <Box component="span" sx={{ display: 'inline-flex', alignItems: 'center', gap: 1 }}>
+                              <Typography component="span" fontWeight="bold">
+                                {item.tokenInfo.name}
+                                {!isNative && <TokenExplorerLink address={item.tokenInfo.address} />}
+                              </Typography>
+                              {isStakingPromoEnabled && item.tokenInfo.type === TokenType.NATIVE_TOKEN && (
+                                <StakeButton tokenInfo={item.tokenInfo} trackingLabel={STAKE_LABELS.asset} plain />
+                              )}
+                              {isEarnPromoEnabled && isEligibleEarnToken(chainId, item.tokenInfo.address) && (
+                                <EarnButton tokenInfo={item.tokenInfo} trackingLabel={EARN_LABELS.asset} plain />
+                              )}
+                            </Box>
+                            <Typography variant="body2" color="primary.light" className={css.mobileBalance}>
+                              <TokenAmount
+                                value={item.balance}
+                                decimals={item.tokenInfo.decimals}
+                                tokenSymbol={item.tokenInfo.symbol}
+                              />
+                            </Typography>
+                          </Stack>
+                        </div>
+                        <Box className={css.mobileValue}>
+                          <Typography>
+                            <FiatBalance balanceItem={item} />
+                          </Typography>
+                          {item.fiatBalance24hChange && (
+                            <Typography variant="caption">
+                              <FiatChange balanceItem={item} inline />
+                            </Typography>
+                          )}
+                        </Box>
+                      </Box>
+                      <Stack direction="row" className={css.mobileButtons}>
+                        <Box className={css.mobileButtonWrapper}>
+                          <SendButton tokenInfo={item.tokenInfo} />
+                        </Box>
+
+                        {isSwapFeatureEnabled && (
+                          <Box className={css.mobileButtonWrapper}>
+                            <SwapButton tokenInfo={item.tokenInfo} amount="0" trackingLabel={SWAP_LABELS.asset} />
+                          </Box>
+                        )}
+                      </Stack>
+                    </Box>
+                  )
+                })}
+          </Box>
+        </Card>
```

Comment:
Seems like a lot of code duplication. Can we achieve the responsiveness via CSS instead? Or at least extract the common parts into separate JSX chunks and reuse them.